bokeh/models/widgets/inputs.py

```diff
@@ -263,3 +263,35 @@ class DatePicker(InputWidget):
     callback = Instance(Callback, help="""
     A callback to run in the browser whenever the current date value changes.
     """)
+
+class TextareaInput(InputWidget):
+    ''' Single-line input widget.
+
+    '''
+
+    value = String(default="", help="""
+    Initial or entered text value.
+    """)
+
+    callback = Instance(Callback, help="""
+    A callback to run in the browser whenever the user unfocuses the TextInput
+    widget by hitting Enter or clicking outside of the text box area.
+    """)
+
+    placeholder = String(default="", help="""
+    Placeholder for empty input field
+    """)
+
+    cols = String(default="", help="""
+    Column count
+    """)
+
+    rows = String(default="", help="""
+    Row count
+    """)
+
+    maxlength = String(default="", help="""
+    Max count of characters in field
+    """)
```

Comment:
Shouldn't `cols`, `rows` and `maxlength` be `Int` properties? Also `max_length` would be more consistent with our existing codebase standards.